app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java

```diff
@@ -135,6 +135,8 @@ else if (aURL.startsWith("resource:") || SessionStore.get().isHomeUri(aURL))
                 aURL = "";
             else if (aURL.startsWith("data:") && SessionStore.get().isCurrentSessionPrivate())
                 aURL = "";
+            else if (aURL.startsWith("javascript:") && aURL.contains(SessionStore.get().getHomeUri()))
```

Comment:
this fixes the bug, but it could trigger false positives. can we check that the previous URL started with the Home URI (`SessionStore.HOME_WITHOUT_REGION_ORIGIN`), and, if so, any URL changes or redirect (e.g., [`"javascript:window.location.replace('" + getHomeUri() + "')"`](https://github.com/MozillaReality/FirefoxReality/blob/81cbd41/app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java#L457)) gets correctly masked (i.e., `aURL remains ""`).

is there a way to set a flag for the active session before the [`loadUri` Gecko call here](https://github.com/MozillaReality/FirefoxReality/blob/81cbd41/app/src/common/shared/org/mozilla/vrbrowser/browser/SessionStore.java#L457) that will tell us "keep the URL empty when we are navigating on the Home page"?

Reply:
> is there a way to set a flag for the active session before the loadUri Gecko call here that will tell us "keep the URL empty when we are navigating on the Home page"?

There is no way of doing that in Gecko ATM.

> this fixes the bug, but it could trigger false positives. can we check that the previous URL started with the Home URI (SessionStore.HOME_WITHOUT_REGION_ORIGIN), and, if so, any URL changes or redirect (e.g., "javascript:window.location.replace('" + getHomeUri() + "')") gets correctly masked (i.e., aURL remains "").

Which are the false positives? Can you give me some example to reproduce? That will only leave the URL empty if it's any javascript URL that includes the home URL.